### DISCUSSION TITLE
Energy prediction shift problem

### DISCUSSION THREAD

**Discussion by kaijiz** [MDIL-SNU/SevenNet#159] [Q&A]
Hi, when i try to apply the sevennet model '7net-0' on your tutorial 'SevenNet tutorial: python interactive' which means i didn't train a new model based on the first part of tutorial but just use your model to perform the 'energy, force, stress prediction' test. I find force and stress prediction performs well but energy prediction has an overall shift compared to VASP, could anyone tell me the potential reason?
<img width="1064" alt="Screen Shot 2025-01-13 at 2 30 58 PM" src="https://github.com/user-attachments/assets/921262b4-e478-4aa2-8706-8f651e6f707b" />

**Comment by hexagonrose** [comment]
Hello! In the tutorial, since we train with a small amount of data and few epochs, we also noticed cases where an energy shift occurred. To improve energy accuracy, you can try the following methods:

1. **Increasing epochs**  
   This is the simplest approach. By continuing the training and reducing the energy loss, the shift should eventually disappear.

2. **Adjusting the loss weight**  
   In the tutorial, `force_loss_weight` is set to 0.2 to reduce the force loss more quickly. You can try setting it to 0.1 to focus more on reducing the energy loss.

Try these methods, and if any issues arise, feel free to ask again!

**Reply by kaijiz** [reply]
Thank you so much for answering. But the model i used is not a new-trained model trained by me. It's one of your mature model '7net-0'. Could you introduce another model or checkpoints which behaves better than this 7net-0 model.
Now, i am thinking the shift arises from the energy correction of Materials Project. Is that right?

**Reply by hexagonrose** [reply]
I misunderstood the question, sorry.
The energy shift doesn't seem to be due to MP's energy correction since SevenNet-0 is trained with uncorrected energy of MPtrj dataset.
Since SevenNet-0 is a pretrained model, it may not fit all systems perfectly, and there could be some errors.
Therefore, trying the larger model, SevenNet-l3i5, or attempting fine-tuning is necessary. According to the fine-tuning tutorial, even with a few epochs, the energy shift improves significantly.

**Reply by kaijiz** [reply]
Hi, if possible, could you tell me how you build the dataset  '[test_md.extxyz](https://github.com/MDIL-SNU/sevennet_tutorial/blob/main/data/evaluation/test_md.extxyz)', i try to build my own dataset by using AIMD simulation and get the trajectory. But the result behaves worse. Because you say Sevennet didn't use correction energy so i really don't know why it has such a large difference. In my example, i just use Tungsten(W) to run the MD.
<img width="1190" alt="Screen Shot 2025-01-16 at 3 46 12 PM" src="https://github.com/user-attachments/assets/ff0e59fe-bd4f-4713-b9d4-cb0e7c6003fe" />

**Reply by hexagonrose** [reply]
It should be fine if you set the DFT calculation to match the settings provided by the Materials Project. Also, it's important to standardize the version of POTCAR (PBE52). 
As mentioned earlier, pretrained models don't always fit every system well. For oxides, which have abundant training data, they often perform well without special tuning. However, for elements like W, where there isn't much training data, errors can be larger. Therefore, try fine-tuning with the MD trajectory data you generated.
